### PR TITLE
add retry on exception and timeout_seconds for get call

### PR DESCRIPTION
**Pull request by dbasunag**
##### Short description: 
rpc error should be retried as during some operations, they are common. However they recoverable. 
We need to add a timeout_seconds for regular get calls, as sometimes, we can have hung call.

### CODE REVIEW

File: ocp_resources/resource.py [hunk 1]
```diff
@@ -334,6 +336,7 @@ def __init__(
         node_selector_labels=None,
         config_file=None,
         context=None,
+        timeout_seconds=60,
```
Comment (myakove) [inline ocp_resources/resource.py:339]: Please give a meaningful name, maybe `api_call_timeout`?

**Review by myakove — CHANGES_REQUESTED** [review]

File: ocp_resources/resource.py (hunk 1 above)
Comment (dbasunag) [reply ocp_resources/resource.py:339]: I have to keep the name, as it is expected by the api to be timeout_seconds.

**Comment by dbasunag** [comment]
@rnetser when I re-requested review from @myakove , I see it removed you as reviewer. I have no idea why and how it happened. Can you please add yourself as a reviewer, as I would prefer to have both your review?

**Review by rnetser — APPROVED** [review]

**Review by rnetser — COMMENTED** [review]
Please fix ci :)

**Comment by dbasunag** [comment]
!verified

**Review by rnetser — APPROVED** [review]

File: ocp_resources/resource.py (hunk 1 above)
Comment (myakove) [reply ocp_resources/resource.py:340]: Then please add docstring, we have `timeout` and `timeout_seconds` that may confuse the user.

**Review by myakove — CHANGES_REQUESTED** [review]

File: ocp_resources/resource.py (hunk 1 above)
Comment (dbasunag) [reply ocp_resources/resource.py:340]: I agree. I will add docstring.

**Review by myakove — APPROVED** [review]
We need to improve the docstring, but we will do it on another PR.

File: ocp_resources/resource.py [hunk 2]
```diff
@@ -498,8 +515,10 @@ def _prepare_resources(cls, dyn_client, singular_name, *args, **kwargs):
 
         get_kwargs = {"singular_name": singular_name} if singular_name else {}
         return dyn_client.resources.get(
-            kind=cls.kind, api_version=cls.api_version, **get_kwargs
-        ).get(*args, **kwargs)
+            kind=cls.kind,
+            api_version=cls.api_version,
+            **get_kwargs,
+        ).get(*args, **kwargs, timeout_seconds=60)
```
Comment (rnetser) [inline ocp_resources/resource.py:521]:
```suggestion
        ).get(*args, **kwargs, timeout_seconds=self.timeout_seconds)
```

**Review by rnetser — CHANGES_REQUESTED** [review]

File: ocp_resources/resource.py (hunk 2 above)
Comment (myakove) [reply ocp_resources/resource.py:521]: :)

**Review by myakove — APPROVED** [review]

File: ocp_resources/resource.py (hunk 2 above)
Comment (dbasunag) [reply ocp_resources/resource.py]: I took the suggestions too fast. _prepare_resources() is a class method so this won't work. and I can't use cls. timeout_seconds here.

**Comment by dbasunag** [comment]
!verified

File: ocp_resources/resource.py (hunk 2 above)
Comment (rnetser) [reply ocp_resources/resource.py:520]: @dbasunag `cls.` should work, you need to define it outside `__init__`

Comment (dbasunag) [reply ocp_resources/resource.py]: Ok, added it as a class attribute with a default value.

**Comment by dbasunag** [comment]
!verified

**Review by rnetser — APPROVED** [review]

**Review by myakove — APPROVED** [review]

**Comment by myakove** [comment]
!verified

**Review by myakove — APPROVED** [review]

**Comment by dbasunag** [comment]
!cherry-pick v4.11

**Comment by dbasunag** [comment]
cherry-pick v4.10

**Pull request merged**